Component: Application State (#1258)
hashicorp/design-system: website/docs/components/application-state/partials/guidelines/guidelines.md

## Usage

### When to use
- When a user's action does not yield any results due to incorrect or incomplete input.
- When an application encounters an issue or error during its operation.

### When not to use
- When the absence of content is expected and does not require an explanation to the user. 
- When there is a clear and intuitive way to add or populate content. 

## Icon

- The icon should be relevant and communicate the same purpose as the errorCode.

## Actions

> - In the footer, you can include two stand-alone links. 

Author: ```suggestion
- In the footer, you can include up to two stand-alone links. 
```

@andgen404 I wonder if we want to reword this just a little bit. WDYT?

Reviewer: @MelSumner I agree with the change

- It’s important to keep these links separate from call-to-action buttons (CTA), as doing so may cause misalignment with the code.

> - For the standalone link, we recommend using the default medium size but use whichever size fits best in the UI.

Reviewer 2: @andgen404 Since the rest of the page is likely to be pretty empty, and we provide them with the styling for the rest of the content within the ApplicationState component, we're already determining the best size for the UI.

```suggestion
- For the standalone link, we recommend using the medium size.
```

> ### Header 
> - The icon should be relevant and communicate the same purpose as the errorCode.
> - The title should be short to provide a clear and concise message.
> 
> 
> ### Body 
> - Focus on relevant information and avoid unnecessary details.
> - Provide a straightforward explanation of the problem or error.
> - Suggestions or guidance for how the user can resolve the issue, if possible.
> 
> ### Footer
> - In the footer, you can include two stand-alone links. It's important to avoid replacing these links with call-to-action buttons (CTA) as doing so may cause misalignment with the code.
> 
> #### Divider (optional)
> Dividers are optional but recommended when dividing content and actions. 

Reviewer: ```suggestion
## Icon 
- The icon should be relevant and communicate the same purpose as the errorCode.

## Actions 
- In the footer, you can include two stand-alone links. 
- It's important to keep these links separate from call-to-action buttons (CTA), as doing so may cause misalignment with the code.

## Content
- The title should be short of providing a clear and concise message. 
- Focus on relevant information and avoid unnecessary details.
- Provide a straightforward explanation of the problem or error.
- Suggestions or guidance for how the user can resolve the issue, if possible. 
```

> - Suggestions or guidance for how the user can resolve the issue, if possible.
> - Provide a straightforward explanation of the problem or error.

Reviewer 2: @andgen404 Updating the order of these based on the order they appear in the component.

```suggestion
- Provide a straightforward explanation of the problem or error.
- Include suggestions or guidance for how the user can resolve the issue, if possible.
```

> - Provide a straightforward explanation of the problem or error.

Reviewer: ```suggestion
- Provide a straightforward explanation of the problem or error.
- Include suggestions or guidance for how the user can resolve the issue, if possible.
```

> - The title should be short of providing a clear and concise message. 
> - Focus on relevant information and avoid unnecessary details.
> - Provide a straightforward explanation of the problem or error.
> - Suggestions or guidance for how the user can resolve the issue, if possible. 

Reviewer: ```suggestion
- The title should be short and provide a clear and concise message.
- Focus on relevant information and avoid unnecessary details.
- Suggestions or guidance for how the user can resolve the issue, if possible.
- Provide a straightforward explanation of the problem or error.
```

Reviewer 2: @andgen404 I think your first suggestion is great ("and provide"), but it makes more sense to have the order of the last two bullet points remain as they are currently (explanation first, then guidance to resolve) as that's the order they happen in the component itself. 

I also recommend adding an action to "Suggestions or guidance for how..." 
Something like: "Include suggestions or guidance for how..."

> - Suggestions or guidance for how the user can resolve the issue, if possible. 

Reviewer: ```suggestion
- Include specific, actionable suggestions or guidance for how the user can resolve the issue.
```

Reviewer: @heatherlarsen I rewrote it with an added action here.

> - Suggestions or guidance for how the user can resolve the issue, if possible. 

Author: ```suggestion
- Include specific, actionable suggestions or guidance for how the user can resolve the issue. 
```

> - Suggestions or guidance for how the user can resolve the issue, if possible. 

Author: ```suggestion
- Include specific, actionable suggestions or guidance for how the user can resolve the issue. 
```

## Usage

### When to use

- When a user's action does not yield any results due to incorrect or incomplete input.
- When an application encounters an issue or error during its operation.

### When not to use

- When the absence of content is expected and does not require an explanation to the user. 
- When there is a clear and intuitive way to add or populate content. 

## Icon

> - The icon should be relevant and communicate the same purpose as the error.

Reviewer: ```suggestion
- The icon, in conjunction with the error, aligns with the purpose of the content and effectively communicates the same message.
```

Reviewer: cc. @MelSumner

## Actions

- In the footer, you can include up to two stand-alone links. 
- It’s important to keep these links separate from call-to-action buttons (CTA), as doing so may cause misalignment with the code.
- For the standalone link, we recommend using the medium size.

> 

Reviewer: ```suggestion
- Use footer actions to redirect or guide users in solving errors/access issues with actionable steps.
```

## Content

- The title should be short and provide a clear and concise message.
- Focus on relevant information and avoid unnecessary details.

> - Include specific, actionable suggestions or guidance for how the user can resolve the issue.
> - Provide a straightforward explanation of the problem or error.

Reviewer: ```suggestion
- Provide a straightforward explanation of the problem or error.
- Include suggestions or guidance for how the user can resolve the issue, if possible.
```

## Usage

### When to use

> - When a user's action does not yield any results due to incorrect or incomplete input.

Reviewer 3: ```suggestion
- When a user’s action does not yield any results due to incorrect or incomplete input.
```

- When an application encounters an issue or error during its operation.

### When not to use

- When the absence of content is expected and does not require an explanation to the user.
- When there is a clear and intuitive way to add or populate content.

## Icon

- The icon should align with the purpose of the content and effectively communicate the same message.

## Actions

- In the footer, you can include up to two stand-alone links. 

> - We don't recommend using buttons, as most actions will navigate the user away from this page. Learn more about [when to use a link vs. a button](https://helios.hashicorp.design/components/link/standalone#usage).

Reviewer 3: ```suggestion
- We don’t recommend using buttons, as most actions will navigate the user away from this page. Learn more about [when to use a link vs. a button](https://helios.hashicorp.design/components/link/standalone#usage).
```

## Usage

### When to use

> Some text here.

Reviewer 3: Would you want to wait until this and other docs content is added before merging to main? (Since this PR has been marked as ready for review.)

Author: I would merge #1264 before merging this one. Would you prefer that I move this back into a draft?

Reviewer 3: I would probably move it back to draft until it's ready to be merged all together. (I wasn't sure if you intended that it be merged first and that those other sections of the doc would be completed afterwards for instance.)

Author: ok, converted back to draft.
